packages/playout-gateway/src/tsrHandler.ts

```diff
@@ -621,15 +627,43 @@ export class TSRHandler {
 			})),
 			new Promise<UpdateDeviceOperationsResult>((resolve) =>
 				setTimeout(() => {
-					const keys = _.keys(promiseOperations)
+					const keys = Object.keys(promiseOperations)
 					if (keys.length) {
-						this.logger.warn(`Timeout in _updateDevices: ${keys.join(',')}`)
+						this.logger.warn(
+							`Timeout in _updateDevices: ${Object.values(promiseOperations)
+								.map((op) => op.deviceId)
+								.join(',')}`
+						)
 					}
-					resolve({
-						success: false,
-						reason: 'timeout',
-						details: keys,
-					})
+
+					Promise.all(
```

Comment:
While we're here, it would be nice to add a comment or two describing what we're doing.
```
// At this point in time, promiseOperations contains the promises that have timed out.
// If we tried to add or re-add a device, that apparently failed so we should remove the device in order to
// give it another chance next time _updateDevices() is called.
```

Reply:
Yeah, good point about explaining the intent here.